### PR TITLE
[ci/testing] only fail ci on failed code coverage upload when not in PR

### PR DESCRIPTION
**Pull request by erdii**
### Check List Before Merging

- [ ] This PR passes all pre-commit hook validations.
- [ ] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

File: .github/workflows/testing.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
       with:
         files: .cache/unit/cover.txt,.cache/integration/cover.txt
         token: ${{ secrets.CODECOV_TOKEN }}
-        fail_ci_if_error: ${{ github.repository_owner == 'package-operator' }}
+        fail_ci_if_error: ${{ github.event_name != 'pull_request' }}
```
Comment (kostola) [inline .github/workflows/testing.yaml:46]: shouldn't it be the opposite?
```yaml
fail_ci_if_error: ${{ github.event_name == 'pull_request' }}
```

Comment (erdii) [reply .github/workflows/testing.yaml:46]: No. CI should NOT fail when this workflow is run during a pull request and this action error.

thus we want:
- for PR -> `fail_ci_if_error: false`
- for main branch tests and other things -> `fail_ci_if_error: true`

Comment (kostola) [reply .github/workflows/testing.yaml:46]: my bad...I misread the title of the PR and thought it was the opposite

**Review by kostola — APPROVED** [review]
It should work...let's see 😜

**Pull request merged**